### PR TITLE
Update DS_2_0_SqStack.cpp

### PR DESCRIPTION
**Pull request by BruceDai003**
The expressions in printf() will get evaluated first before print out, so the counter needs to be put outside the printf() function.

### CODE REVIEW

File: DataStructure/Src/DS_2_StackAndQueue/DS_2_0_SqStack.cpp [hunk 1]
```diff
@@ -120,7 +120,8 @@ int GetTopOther1(SqStack S) {
 void PrintStack(SqStack S) {
     printf("从栈顶元素开始，栈如下：\n");
     while (S.top >= 0) {//注意判空的条件
-        printf("S[%d]=%d\n", S.top, S.data[S.top--]);
```
Comment (KimYangOfCat) [inline DataStructure/Src/DS_2_StackAndQueue/DS_2_0_SqStack.cpp:123]: > The expressions in printf() will get evaluated first before print out, so the counter needs to be put outside the printf() function.

Thanks for your PR~
But the effect of `printf("S[%d]=%d\n", S.top, S.data[S.top--])` and ` printf("S[%d]=%d\n", S.top, S.data[S.top]);S.top--` is the same ~ 
Maybe this way of writing is a little hard to understand，their execution results are the same. 
Thanks~

Comment (BruceDai003) [reply DataStructure/Src/DS_2_StackAndQueue/DS_2_0_SqStack.cpp:123]: I can't agree. I thought they would produce the same results. But I compiled the code, the results are different. So I think in the printf() function, all expressions are getting evaluated before printed out, including S.top--. Could you try to compile this?

Comment (KimYangOfCat) [reply DataStructure/Src/DS_2_StackAndQueue/DS_2_0_SqStack.cpp:123]:
```c
int main() {
//    testModule();
    int arr[5]={0,1,2,3,4};
    printf("-----n1-----\n");
    int n1=2;
    printf("before:%d\n",arr[n1]);
    printf("run:%d\n",arr[n1--]);
    printf("after: %d \n",n1);

    printf("-----n3-----\n");
    int n3=2;
    printf("before:%d\n",arr[n3]);
    printf("run:%d\n",arr[n3]);
    n3--;
    printf("after: %d \n",n3);

    printf("-----n2-----\n");
    int n2=2;
    printf("before:%d\n",arr[n2]);
    printf("run:%d\n",arr[--n2]);
    printf("after: %d \n",n2);

    printf("-----n4-----\n");
    int n4=2;
    printf("before:%d\n",arr[n4]);
    --n4;
    printf("run:%d\n",arr[n4]);
    printf("after: %d \n",n4);
    return 0;
}
```
You can run the above code to verify the results. I have run the code and I got the following result：
image

Comment (BruceDai003) [reply DataStructure/Src/DS_2_StackAndQueue/DS_2_0_SqStack.cpp:123]: Please take a note, in printf function, there is no guarantee as when S.top or S.data[S.top--] will get evaluated first. This is called an undefined behaviour in C/C++, which means different compilers will probably get different results, even the same compiler will get different results in different times. 
Check this out: https://stackoverflow.com/questions/30452526/using-unary-increment-decrement-in-printf-function
https://stackoverflow.com/questions/949433/why-are-these-constructs-using-pre-and-post-increment-undefined-behavior? Especially the third answer.

**Review by KimYangOfCat — COMMENTED** [review]
Thanks for your explanation, I don’t know the concept of undefined behavior, but new I got it ~ Thanks again！

File: DataStructure/Src/DS_2_StackAndQueue/DS_2_0_SqStack.cpp (hunk 1 above)
Comment (KimYangOfCat) [reply DataStructure/Src/DS_2_StackAndQueue/DS_2_0_SqStack.cpp:123]: Thanks for your explanation, I don’t know the concept of undefined behavior, but new I got it ~ Thanks again！

**Pull request merged**